<launch>

    <arg name="arm_gui" default="true"/>
    <arg name="rviz" default="true"/>

    <include file="$(find victor_description)/launch/visualize_victor.launch">
        <arg name="gui" value="false"/>
        <arg name="joint_state_publisher" value="false"/>
        <arg name="rviz" value="$(arg rviz)"/>
    </include>

    <group ns="left_arm">
        <node pkg="victor_hardware_interface" type="victor_hardware_interface_arm_wrapper_node" name="left_arm_lcm_bridge_node" required="true" output="screen" >
            <param name="send_lcm_url" type="string" value="udp://10.10.10.12:30000" />
            <param name="recv_lcm_url" type="string" value="udp://10.10.10.10:30002" />
        </node>
    </group>

    <group ns="right_arm">
        <node pkg="victor_hardware_interface" type="victor_hardware_interface_arm_wrapper_node" name="right_arm_lcm_bridge_node" required="true" output="screen" >
            <param name="send_lcm_url" type="string" value="udp://10.10.10.11:30000" />
            <param name="recv_lcm_url" type="string" value="udp://10.10.10.10:30001" />
        </node>
    </group>

    <node pkg="victor_hardware_interface" type="joint_state_publisher.py" name="joint_state_publisher"/>
    <node pkg="victor_hardware_interface" type="arm_gui.py" name="arm_gui" if="$(arg arm_gui)"/>
</launch>

